stdlib/public/core/StringBridge.swift

```diff
@@ -234,35 +243,143 @@ internal func _bridgeCocoaString(_ cocoaString: _CocoaString) -> _StringGuts {
     //   2) If it's mutable with no associated information, then a copy must
     //      happen; might as well eagerly bridge it in.
     //   3) If it's mutable with associated information, must make the call
-    let immutableCopy
-      = _stdlib_binary_CFStringCreateCopy(cocoaString) as AnyObject
+    let immutableCopy =
+      _stdlib_binary_CFStringCreateCopy(cocoaString) as AnyObject
     
-#if !(arch(i386) || arch(arm))
+    #if !(arch(i386) || arch(arm))
     if _isObjCTaggedPointer(immutableCopy) {
       return _StringGuts(_SmallString(taggedCocoa: immutableCopy))
     }
-#endif
+    #endif
     
     let (fastUTF8, isASCII): (Bool, Bool)
     switch _getCocoaStringPointer(immutableCopy) {
     case .ascii(_): (fastUTF8, isASCII) = (true, true)
     case .utf8(_): (fastUTF8, isASCII) = (true, false)
     default:  (fastUTF8, isASCII) = (false, false)
     }
-    let length = _stdlib_binary_CFStringGetLength(immutableCopy)
     
     return _StringGuts(
       cocoa: immutableCopy,
       providesFastUTF8: fastUTF8,
       isASCII: isASCII,
       length: length)
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeConstantCocoaString(
+  _ cocoaString: __owned AnyObject, _ length: Int) -> String {
+  let (fastUTF8, isASCII): (Bool, Bool)
+  switch _getCocoaStringPointer(cocoaString) {
+  case .ascii(_): (fastUTF8, isASCII) = (true, true)
+  case .utf8(_): (fastUTF8, isASCII) = (true, false)
+  default:  (fastUTF8, isASCII) = (false, false)
   }
+  
+  return String(_StringGuts(
+    cocoa: cocoaString,
+    providesFastUTF8: fastUTF8,
+    isASCII: isASCII,
+    length: length))
+}
+
+@_effects(releasenone)
+public //SPI(Foundation)
+func _bridgeUnknownCocoaString(
+  _ cocoaString: __owned AnyObject, _ length: Int) -> String {
+ return _bridgeConstantCocoaString(cocoaString, length)
+}
+
+@_effects(readonly)
+public //SPI(Foundation)
+func _isRebridgedSwiftStringClass(_ cls: AnyClass) -> Bool {
```

Comment:
This is sort of an awful way to avoid making __StringStorage and __SharedStringStorage's classes public… I think I was worried about causing a retain of the NSString to pass it into _rebridgeSwiftString() as __owned, but we *do* want to pass it __owned if we're definitely going to do it. Hm.